src/services/earn/earn-service.ts

```diff
@@ -475,6 +476,110 @@ export class EarnService implements IEarnService {
     return buildCompanionMulticall({ chainId, calls, value: increaseInfo.value });
   }
 
+  async buildClaimDelayedWithdrawPositionTx({
+    chainId,
+    positionId,
+    recipient,
+    permissionPermit,
+    claim,
+  }: {
+    chainId: ChainId;
+    positionId: PositionId;
+    recipient: Address;
+    permissionPermit?: EarnPermissionPermit;
+    claim: { tokens: { token: TokenAddress; convertTo?: TokenAddress }[]; swapConfig?: EarnActionSwapConfig };
+  }): Promise<BuiltTransaction> {
+    const vault = EARN_VAULT.address(chainId);
+    const manager = DELAYED_WITHDRAWAL_MANAGER.address(chainId);
+    const [, , tokenId] = positionId.split('-');
+    const bigIntPositionId = BigInt(tokenId);
+
+    // Get withdrawable funds for each token to convert
+    const withdrawableFunds = await this.providerService.getViemPublicClient({ chainId }).multicall({
+      contracts: claim.tokens.map(({ token }) => ({
+        address: manager,
+        abi: delayerWithdrawalManagerAbi,
+        functionName: 'withdrawableFunds',
+        args: [bigIntPositionId, token as ViemAddress],
+      })),
+      allowFailure: false,
+    });
+
+    const claimWithFunds = claim.tokens
+      .map((token, index) => ({ ...token, amount: withdrawableFunds[index] as bigint }))
+      .filter(({ amount }) => amount > 0n);
+    if (claimWithFunds.length == 0) {
+      throw new Error('No funds to claim');
+    }
+    const claimsToConvert = claimWithFunds.filter(({ convertTo }) => !!convertTo);
+
+    if (claimsToConvert.length == 0) {
+      // If don't need to convert anything, then just call the delayer withdrawal manager
+      const calls = claimWithFunds.map(({ token }) =>
+        encodeFunctionData({
+          abi: delayerWithdrawalManagerAbi,
+          functionName: 'withdraw',
+          args: [bigIntPositionId, token as ViemAddress, recipient as ViemAddress],
+        })
+      );
+
+      return {
+        to: manager,
+        data: encodeFunctionData({
+          abi: delayerWithdrawalManagerAbi,
+          functionName: 'multicall',
+          args: [calls],
+        }),
+      };
+    }
+
+    // If we get to this point, then we'll use the Companion for swap & transfer
+    const calls: Hex[] = [];
+
+    // Handle permission permit
+    if (permissionPermit) {
+      calls.push(buildPermissionPermit(bigIntPositionId, permissionPermit, vault));
+    }
+
+    // Handle claim
+    calls.push(
+      ...claim.tokens.map(({ token, convertTo }) =>
```

Comment:
We should use this, right?
```suggestion
      ...claimWithFunds.map(({ token, convertTo }) =>
```